src/main/java/jenkins/bouncycastle/api/PEMManager.java

```diff
@@ -0,0 +1,402 @@
+    public PublicKey toPublicKey() {
+        if (object instanceof PublicKey) {
+            return (PublicKey) object;
+        } else if (object instanceof KeyPair) {
+            return ((KeyPair) object).getPublic();
+        } else if (object instanceof Certificate) {
+            return ((Certificate) object).getPublicKey();
+        }
+        return null;
+    }
+
+    /**
+     * Obtain {@link Certificate} object from the read PEM.
+     * 
+     * @return {@link Certificate} with the certificate, null if a certificate could not be obtained from the current
+     * data
+     */
+    @Nullable
+    public Certificate toCertificate() {
+        if (object instanceof Certificate) {
+            return ((Certificate) object);
+        }
+        return null;
+    }
+
+    /**
+     * Obtain {@link PrivateKey} object from the read PEM. If the PEM data contained other object type like
+     * {@link KeyPair}, the private key will be extracted from them.
+     * 
+     * @return {@link PrivateKey} with the private key, null if a private key could not be obtained from the current
+     * data
+     */
+    @Nullable
+    public PrivateKey toPrivateKey() {
+        if (object instanceof PrivateKey) {
+            return (PrivateKey) object;
+        } else if (object instanceof KeyPair) {
+            return ((KeyPair) object).getPrivate();
+        }
+        return null;
+    }
+
+    /**
+     * Obtains raw JCA or BouncyCastle {@link Object} from the read PEM. Depending on the PEM nature or the object
+     * passed to the {@link #PEMManager(Object pemObject)}, the returned object can be one of the following (not
+     * exhaustive list) and any classes that inherit from them:
+     * <ul>
+     * <li><strong>JCA</strong>
+     * <ul>
+     * <li>{@link Certificate}
+     * <li>{@link java.security.cert.CRL}
+     * <li>{@link KeyPair}
+     * <li>{@link PublicKey}
+     * <li>{@link PrivateKey}
+     * </ul>
+     * </ul>
+     * <ul>
+     * <li><strong>Bouncy Castle</strong>
+     * <ul>
+     * <li>{@link org.bouncycastle.asn1.cms.ContentInfo}
+     * <li>{@link org.bouncycastle.jce.spec.ECNamedCurveParameterSpec}
+     * <li>{@link org.bouncycastle.pkcs.PKCS10CertificationRequest}
+     * <li>{@link org.bouncycastle.jce.provider.X509CertificateObject}
+     * <li>{@link org.bouncycastle.x509.X509V2AttributeCertificate}
+     * </ul>
+     * </ul>
+     * 
+     * @return {@link Object} read from the PEM
+     */
+    @Nullable
+    public Object getRawObject() {
+        return object;
+    }
+
+    /**
+     * Obtains the fingerprint of the private key in the "ab:cd:ef:...:12" format, which basically is an SHA1 digest
+     * from the key, encoded in hex format.
+     * 
+     * @return private key fingerprint in hex format "ab:cd:ef:...:12", null if the private key could not be obtained
+     * from the current PEM data.
+     * @throws IOException thrown if a problem exists creating the fingerprint
+     */
+    @Nullable
```

Comment:
this is @CheckForNull not @Nullable as the caller does not know before hand whether the PEM encoded object is a certificate or a key... thus a priori we do not know whether this could return null or non-null so the caller cannot be expected to know the null-ness and this means it cannot be @Nullable and must be @CheckForNull